app/controllers/api/teams_controller.rb

```diff
@@ -35,27 +33,10 @@ def create
     render_json team
   end
 
-  # PATCH /api/teams/:id
-  def update
-    authorize team
-    team.update!(model_params)
-
-    add_info(t('flashes.teams.updated'))
-
-    render_json
-  end
-
-  # DELETE /api/teams/:id
-  def destroy
-    authorize team
-    team.destroy
-    render_json
-  end
-
   private
 
   def fetch_entries
-    ::Teams::FilteredList.new(current_user, params).fetch_entries
+    @entries ||= ::Teams::FilteredList.new(current_user, params).fetch_entries
```

Comment:
we need to cache this since you're using this for the policy check on line 19 and render it afterwards!
this would trigger the whole sql twice ! performance !